ttnn/cpp/ttnn/operations/conv/conv2d/device/kernels/reader_conv_activations_padded_with_halo_3x3_weights_v2.cpp

```diff
@@ -120,25 +136,61 @@ void kernel_main() {
             uint32_t reader_offset = act_l1_read_addr + (reader_offsets[reader_offset_idx] * conv_act_c_read_bytes);
             // #pragma GCC unroll 4 // unroll didn't help, but act_block_h_datums (loop bound) being const does help
             uint32_t act_block_h_datums_read_curr = bh == act_num_blocks_h - 1 ? act_block_h_datums_read_last_block : act_block_h_datums_read;
-            for (uint32_t bhd = 0; bhd < act_block_h_datums_read_curr; bhd++) {
-                // local read from reader_index + reader_offset;
-                #ifdef SPLIT_READER
-                uint32_t two_reader_indices = cache_packed_reader_indices ? local_packed_reader_indices[bhd] : packed_reader_indices_ptr[reader_idx];
-                #else // no split reader
-                uint32_t two_reader_indices = packed_reader_indices_ptr[reader_idx];
-                #endif
-                uint32_t reader_idx_1 = two_reader_indices & 0xffff;
-                uint32_t reader_idx_2 = two_reader_indices >> 16;
-
-                act_l1_offset = reader_offset + (reader_idx_1 * conv_act_c_read_bytes);
-                noc_async_read_one_packet_with_state<true>(act_l1_offset, l1_write_addr_act);
-                l1_write_addr_act += (coalesced_read_bytes + act_block_w_extra_align_bytes);
-
-                act_l1_offset = reader_offset + (reader_idx_2 * conv_act_c_read_bytes);
-                noc_async_read_one_packet_with_state<true>(act_l1_offset, l1_write_addr_act);
-                l1_write_addr_act += (coalesced_read_bytes + act_block_w_extra_align_bytes);
-
-                reader_idx++;
+            if(dilation_w==1)
```

Comment:
This much code should not be duplicated like this.

You could leave the common code as is and use `#ifdef` if necessary to split out the non common code.

Although ideally it would be one code path, with the variable values changing depending on the dilation.

Reply:
Cleaned up.